pkg/kubelet/images/image_manager.go

```diff
@@ -136,12 +185,32 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, pod *v1.Pod, conta
 	}
 
 	present := imageRef != ""
-	if !shouldPullImage(container, present) {
+
+	var pulledBySecret, ensuredBySecret bool
+	if present {
+		pulledBySecret, ensuredBySecret, err := m.isEnsuredBySecret(imageRef, spec, pullSecrets)
+		if err != nil {
+			klog.V(4).ErrorS(err, "Error get ensured check by secret", "image", image, "imageRef", imageRef, "pulledBySecret", pulledBySecret, "ensuredBySecret", ensuredBySecret)
```

Comment:
Should we return the error to the caller instead of just logging?

Reply:
Yes.